lib/rules/no-restricted-imports.js

```diff
@@ -452,6 +536,24 @@ module.exports = {
 		function reportPathForPatterns(node, group, importNames) {
 			const importSource = node.source.value.trim();
 
+			// Skip if this is a type-only import and it's allowed
+			if (
+				node.type === "ImportDeclaration" &&
+				isTypeOnlyImport(node) &&
+				group.allowTypeImports
+			) {
+				return;
+			}
+
+			// Skip if this is a type-only export and it's allowed
+			if (
+				node.type === "ExportNamedDeclaration" &&
+				isTypeOnlyExport(node) &&
+				group.allowTypeImports
```

Comment:
[Testing] Removing this `&& group.allowTypeImports` didn't fail any unit tests. I added one locally that was a copy of the test that fails if the earlier `&& group.allowTypeImports` is removed, `patterns: ["@typescript-eslint/utils/dist/*"],`, but switched to an export instead of an import.